arangod/Aql/DocumentProducingHelper.cpp

```diff
@@ -286,6 +286,10 @@ bool DocumentProducingFunctionContext::checkFilter(
 }
 
 bool DocumentProducingFunctionContext::checkFilter(ExpressionContext& ctx) {
+  if (ADB_UNLIKELY(_numFiltered % 1000 == 0 && _query.killed())) {
```

Comment:
The `% 1000` will end up as an IDIV instruction with considerable latency.
Can we turn this into a `% 1024` so the compiler will turn this into a much simpler instruction?